api/web3server.go

```diff
@@ -104,7 +105,7 @@ func NewWeb3Handler(core CoreService, cacheURL string, batchRequestLimit int) We
 }
 
 // HandlePOSTReq handles web3 request
-func (svr *web3Handler) HandlePOSTReq(ctx context.Context, reader io.Reader, writer apitypes.Web3ResponseWriter) error {
+func (svr *web3Handler) HandlePOSTReq(ctx context.Context, reader io.Reader, writer apitypes.Web3ResponseWriter) (err error) {
```

Comment:
no need to add `err` since it'll be defined at L111?

Reply:
It's used for defer func, only in this way you can actually change the returned error in the defer function